.github/workflows/ci.yml

```diff
@@ -1,134 +1,192 @@
+          - macOS
+          - windows
+        toolchain:
+          - stable
+          - nightly
+    runs-on: ${{ matrix.os }}-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@v1
         with:
-          command: test
-          args: --doc --features testing-helpers
+          toolchain: ${{ matrix.toolchain }}
+      - run: rustup default ${{ matrix.toolchain }}
 
+      - run: cargo test --features testing-helpers
 
-  clippy:
-    name: Clippy
+  test-features:
+    name: test features
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@v1
         with:
-          profile: minimal
           toolchain: nightly
-          components: clippy
-          override: true
+      - run: rustup default nightly
 
-      - name: Clippy
-        uses: actions-rs/cargo@v1
-        with:
-          command: clippy
-          args: --all-features -- -D warnings
+      - name: Install tomljson
+        run: go get github.com/pelletier/go-toml/cmd/tomljson
 
+      - run: ci/test_all_features.sh
+        env:
+          GOBIN: "$HOME/go/bin"
+          PATH: "$PATH:$GOBIN"
 
-  rustfmt:
+
+
+
+  #################
+  # Documentation #
+  #################
+
+  rustdoc:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@v1
         with:
-          profile: minimal
           toolchain: nightly
-          components: rustfmt
-          override: true
 
-      - name: rustfmt
-        uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: --all -- --check
+      - run: cargo +nightly doc
+        env:
+          RUSTDOCFLAGS: --cfg docsrs
 
 
-  msrv:
-    # TODO: Use MSRV once it understands new `Cargo.lock` format.
-    # name: MSRV
-    name: Minimal crates versions
-    strategy:
-      fail-fast: false
-      matrix:
-        # msrv: ["1.36.0"]
-        msrv: [stable]
-        os: [ubuntu, macOS, windows]
-    runs-on: ${{ matrix.os }}-latest
+
+
+  #############
+  # Releasing #
+  #############
+
+  release-crate:
+    name: release on crates.io
+    if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+    needs: ["release-github"]
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@v1
         with:
-          profile: minimal
-          toolchain: nightly
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: ${{ matrix.msrv }}
-          override: true
+          toolchain: stable
+
+      - name: Publish `derive_more` crate
+        run: cargo publish -p derive_more
+                           --token ${{ secrets.CRATESIO_TOKEN }}
```

Comment:
I prefer to do the `cargo publish` command on my local machine, instead of from CI. Mainly because of the security implications of storing my CRATESIO_TOKEN in github actions, and secondly because `cargo release` makes all this very easy anyway to do from my local machine.

Reply:
@JelteF as for me, the easiest way is just to push the Git tag and don't bother 🙃  

> Mainly because of the security implications of storing my CRATESIO_TOKEN in github actions

I'm not aware of such. Moreover, I would argue that keeping such token in GitHub secrets is more safe than keeping it on a local machine (or being authenticated locally on `crates.io`), as the latter is more exposed to untrusted environments. So I tend to not keep any credentials locally asap. Usually, I create a separate token for separate projects (named after the project), so any token could be easily revoked at any time on `crates.io`.

> I prefer to do the `cargo publish` command on my local machine, instead of from CI.

Doing so, I also wanted to streamline the release process for any person who would be theoretically allowed to do the release.

Reply:
Done in https://github.com/JelteF/derive_more/pull/212/commits/e6a9c6f4d23a668dd3d3e629e45b5bc79146a898.

Reply:
Thanks for removing the release job. Regarding the security implications:

> Warning: Any user with write access to your repository has read access to all secrets configured in your repository. 

Source: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions

While I trust you (and possible future contributors). Giving access to my crates.io secrets to people I've only met over the internet is something I'd rather not do.